### DISCUSSION TITLE
0054 XLS-54d: Add DestinationTag to NFTokenCreateOffer transaction and NFTokenOffer object

### DISCUSSION THREAD

**Discussion by florent-uzio** [XRPLF/XRPL-Standards#152] [XLS Proposals]
## The earlier version can be found by looking at the edit history.

**Comment by scottschurr** [comment]
This proposal solves a problem for those folks who custody NFTs for others.  And I think it's a reasonable approach.  This approach also has the advantage that it works very much like `DestinationTag`s for Payments, which is how folks custody fungible tokens on ledger.  Using the same tool to solve a similar problem makes it easier for everyone to think about.

Something to pay attention to in the implementation is that this interacts with the `lsfRequireDestTag` flag on an `AccountRoot`.  The `NFTokenCreateOffer` and `NFTokenAcceptOffer` transactors should enforce the behavior of that flag.  That detail is worth adding to the spec.

The details of enforcing the flag will be fiddly.  You'll need to think through both kinds of offers:

- `NFTokenBuyOffer` and
- `NFTokenSellOffer`.

You'll also need to think through all three ways of accepting offers:

- Accepting a buy offer,
- Accepting a sell offer, and
- Brokering a buy and a sell offer.

I had some initial thoughts on what the rules should be, but they are more complicated than I initially thought.  So it's better that you puzzle them out than have me leading you down the wrong path.

[a comment by an automated account is not shown]
